cli/src/movement_util.rs

```diff
@@ -122,9 +158,11 @@ fn get_target_commit(
         [target] => target,
         [] => {
             // We found no ancestor/descendant.
-            return Err(user_error(
-                direction.target_not_found_message(change_offset),
-            ));
+            return Err(user_error(direction.target_not_found_message(
+                edit,
+                change_offset,
+                &short_change_hash(start_commit.change_id()),
```

Comment:
nit: how about passing the `&ChangeId` into the function? that makes the signature clearer

Reply:
or use commit summary template?

Reply:
> nit: how about passing the &ChangeId into the function? that makes the signature clearer

I've done this.

> or use commit summary template?
I don't know what this is. Any pointers? Thanks

Reply:
Resolving. Do re-open if something is not quite right. Thanks.

> > nit: how about passing the &ChangeId into the function? that makes the signature clearer
> 
> I've done this.
> 
> > or use commit summary template?

Whoops! My previous comment quoted wrongly and hid my question:

> > or use commit summary template?

I don't know what this is. Any pointers? Thanks

Reply:
It's these functions: https://github.com/martinvonz/jj/blob/bb018a54c311cc7d881dc8352d6ef100902b9096/cli/src/cli_util.rs#L1119-L1147

Reply:
I think this PR already seems like an improvement, but I agree with Yuya that printing the commit summary seems even better, as long as we can lay it out in reasonable way when there are multiple parents. We'll presumably want to print one per line, which means we might want something like this:
```
Error: No other descendant found 1 commit(s) forward from:
  xxx aaa | parent 1
  yyy bbb | parent 2
```

FYI, the `ErrorHint` type has a `Formatted` variant for capturing color.

Reply:
```
$ jj next
Error: No other descendant found 1 commit(s) forward from:
Hint: ynsslwsp ea0238fc (empty) b2-c3
Hint: skrvwlnk 4609a180 (empty) b1-c3
```

> It's these functions:
> 
> https://github.com/martinvonz/jj/blob/bb018a54c311cc7d881dc8352d6ef100902b9096/cli/src/cli_util.rs#L1119-L1147

Thanks!

Reply:
> I think this PR already seems like an improvement, but I agree with Yuya that printing the commit summary seems even better, as long as we can lay it out in reasonable way when there are multiple parents. We'll presumably want to print one per line, which means we might want something like this:
> 
> ```
> Error: No other descendant found 1 commit(s) forward from:
>   xxx aaa | parent 1
>   yyy bbb | parent 2
> ```
> 
> FYI, the `ErrorHint` type has a `Formatted` variant for capturing color.

So this actually has a 'Hint:' prefix. is that okay? So it looks like so:

 ```
 Error: No other descendant found 1 commit(s) forward from:
   Hint: xxx aaa | parent 1
   Hint: yyy bbb | parent 2
 ```

See the updated test output for details.

Reply:
Perhaps, the error message can be split to the error and hint lines? something like
```
Error: No other descendant found N commits forward from the working copy parent
Hint: Working copy parent: xxx aaa | ...
```

Reply:
Much better. I've gone ahead to structure things like this. Thanks!